gwtbootstrap3/src/main/java/org/gwtbootstrap3/client/ui/base/mixin/DataTargetMixin.java

```diff
@@ -34,15 +35,28 @@
  */
 public class DataTargetMixin<T extends UIObject & HasDataTarget> extends AbstractMixin implements HasDataTarget {
 
+    /**
+     * Ensures the given element has a non-empty id.
+     *
+     * @param element The element being examined.
+     * @see Document#createUniqueId()
+     */
+    public static void ensureId(final Element element) {
```

Comment:
Why this method is ```public static``` ? If you want it to be public and static, like a helper method, it should be moved to an helper class. Here, you should declare it as ```private void ensureId()```.